app/src/main/AndroidManifest.xml

```diff
@@ -77,6 +78,17 @@
         <activity
             android:name=".HiltTestActivity"
             android:exported="false" />
+
+        <provider
+            android:name="androidx.core.content.FileProvider"
+            android:authorities="com.palfinder.fileprovider"
+            android:exported="false"
+            android:grantUriPermissions="true">
+            <meta-data
+                android:name="android.support.FILE_PROVIDER_PATHS"
+                android:resource="@xml/file_paths" />
+        </provider>
```

Comment:
What does this do exactly, could you explain me?

Reply:
It is the new way to access the local files on an android phone, as explained here : https://developer.android.com/reference/android/support/v4/content/FileProvider